### PR TITLE
Feature/promo enrollment

### PR DESCRIPTION
**Pull request by lafbelmonte**
-added graphql mutation to enroll a member to a promo
-added email, bankaccount, and balance field on member
-interfaced promo for signup and deposit
-required password only for creation of member (removed on update)

### CODE REVIEW

File: koa-crud/src/use-cases/promo-enrollments/enroll-to-promo.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  membersStore: MembersStore;
+  promosStore: PromosStore;
+  R;
+  promoEnrollmentsStore: PromoEnrollmentsStore;
+}): UseCase<boolean> => {
+  const underscoreToCamelCase = (str) => {
+    return str.toLowerCase().replace(/_+(\w|$)/g, ($$, $1) => {
+      return $1.toUpperCase();
+    });
+  };
```
Comment (ralphcasipe1) [inline koa-crud/src/use-cases/promo-enrollments/enroll-to-promo.ts]: You can also use this, https://github.com/sindresorhus/camelcase?

File: koa-crud/src/graphql/typeDefs/promo-enrollments.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export default gql`
+  extend type Mutation {
+    enrollToPromo(input: EnrollToPromoInput!): Boolean
+  }
+
+  input EnrollToPromoInput {
+    promo: String!
+  }
+`;
```
Comment (ralphcasipe1) [inline koa-crud/src/graphql/typeDefs/promo-enrollments.ts]: If there is only one argument at best then you can directly put it. No need to add an `input`. You can do it like this:
```gql
extend type Mutation {
    enrollToPromo(promo: ID!): Boolean!
}
```

**Review by ralphcasipe1 — CHANGES_REQUESTED** [review]
For the mean time `ref`, `populate` and `lookup`. Later on, we will just manually reference it. Mostly, this is the case in cqrs where read models should be lean as much as possible.

File: koa-crud/src/graphql/typeDefs/promo-enrollments.ts (hunk 2 above)
Comment (lafbelmonte) [reply koa-crud/src/graphql/typeDefs/promo-enrollments.ts]: Noted sir

File: koa-crud/src/use-cases/promo-enrollments/enroll-to-promo.ts (hunk 1 above)
Comment (lafbelmonte) [reply koa-crud/src/use-cases/promo-enrollments/enroll-to-promo.ts]: noted sir

**Review by ralphcasipe1 — APPROVED** [review]
LGTM!

**Pull request merged**